source/declarations.tex

```diff
@@ -2521,34 +2520,24 @@
 
 \pnum
 Thus, a declaration of a particular identifier has the form
-\begin{codeblock}
-T D
-\end{codeblock}
-where
-\tcode{T}
+\tcode{T} \tcode{D} where \tcode{T}
 is of the form \opt{\grammarterm{attribute-specifier-seq}}
 \grammarterm{decl-specifier-seq}
-and
-\tcode{D}
-is a declarator.
+and \tcode{D} is a declarator.
 Following is a recursive procedure for determining
 the type specified for the contained
 \grammarterm{declarator-id}
 by such a declaration.
 
 \pnum
-First, the
-\grammarterm{decl-specifier-seq}
-determines a type.
-In a declaration
-\begin{codeblock}
-T D
-\end{codeblock}
-the
-\grammarterm{decl-specifier-seq}
-\tcode{T}
-determines the type
-\tcode{T}.
+First, the type \tcode{T} is determined.
```

Comment:
We have two different things we're calling `T` here: a *decl-specifier-seq* and a type. This rewrite makes it less clear that that's happening by removing the "the *decl-specifier-seq* `T` determines the type `T`" part. I think we should take this a bit further. How about replacing the previous paragraph ("Thus[...]") with:

> Thus, a declaration of a particular identifier has the form *attribute-specifier-seq*[opt] *decl-specifier-seq* *declarator*. The optional *attribute-specifier-seq* appertains to the declared entity. Following is a recursive procedure for determining the type of the declared entity.

Then we only have one `T`, and it's a type, not a *decl-specifier-seq*. So we can start this second paragraph with something like:

> First, a type `T` is determined from the *decl-specifier-seq*. [Example: ...

Reply:
Perhaps we want to talk about the type of the *declarator-id* rather than the declared entity? Declarations introduce names (which denote entities), after all.